src/machine/machine_nrf52xxx.go

```diff
@@ -183,7 +183,12 @@ func (a *ADC) Get() uint16 {
 		resolutionAdjustment = 4 // 12bit
 	}
 
-	return rawValue.Get() << resolutionAdjustment
+	value := int16(rawValue.Get())
+	if value < 0 {
+		value = 0
+	}
+
+	return uint16(value << resolutionAdjustment)
```

Comment:
@deadprogram we do return `uint16` in the end, if that's you after.

Reply:
I've just repeated exact logic as we had before: casting register value to `int16`, checking for `0` and then casting to `uint16` after applying resolution adjustment